docs/guides/migration.md

```diff
@@ -0,0 +1,223 @@
+---
+subcategory: "Guides"
+page_title: "Migration"
+description: |-
+    An overview of resource and datasource migration changes
+---
+
+## Introduction
+
+Since the first release in September 2020, the Terraform ACI provider has come a long way, adding new resources and datasources to make managing Cisco ACI environments easier. Over the years, we've moved from SDKv1 to [SDKv2](https://developer.hashicorp.com/terraform/plugin/sdkv2), and now to the latest [Terraform Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework), which is the recommended way to develop Terraform plugins and offers many advantages over SDKv2.
+
+To take full advantage of the new features in the Terraform Plugin Framework, we had to completely rewrite the Terraform ACI provider resources and datasources. While this was a big task, it also gave us a chance to review and improve the current ACI provider. This guide explains the new features that the Terraform Plugin Framework provides and the changes we have made to the ACI provider and why we made them.
+
+## Upgrading the ACI Terraform Provider
+
+Upgrading the ACI Terraform provider to a new version involves several steps to ensure a smooth transition and to avoid any disruptions to your existing configurations. This guide will walk you through the process, including taking a backup of your current state.
+
+#### Step 1: Backup Your Current State
+
+Before making any changes, it's crucial to back up your current Terraform state. This ensures that you can restore your environment if anything goes wrong during the upgrade process.
+
+1. **Local Backend**: Navigate to Your Terraform Project Directory and open a terminal and navigate to the directory containing your Terraform configuration files and copy the current state file (`terraform.tfstate`) to a backup location.
```

Comment:
Navigate to Your Terraform Project Directory and open a terminal and navigate to the directory containing your Terraform configuration files and copy the current state file (`terraform.tfstate`) to a backup location.
->
Open a terminal and navigate to the directory containing your ACI Terraform configuration files with the state and copy the current state file (`terraform.tfstate`) to a backup location.